hfaerber/Refactor-tractor-Fit-lit
# Fetch continuation and refactoring date and user selection

#### What's this PR do?

1) 
Refactors how the user is being selected.  
No the random number is matching with the user id in the dataset of all users.  
This will allow us to implement a login screen to select a specific user later.

Additional refactoring could be done to eliminate where the logic is tied to the user.id always being the index+1 but since this works with our previous datasets as well as our current api datasets, we can postpone this until after higher priority tasks are completed.  
These items are noted in issue #17 

2) 
Refactors the date so that it is always using the actual live date (today) instead of the last date found in the activity dataset.  

This is important now that we are getting our dataset from an Api that can have additional dates posted to it by classmates.  

The Api's dataset has dates previous to and after today's date.  
Anything posted by classmates will go onto the end of the dataset so using today's date should keep us safe from any surprises in the data.

3) 
Reorganizes the code in scripts so that essentially everything is within the .then() statements of our api fetches.  

The promise being returned from the fetch wasn't accessible (even when saving to global variable) outside the .then block so everything we want to do to it/with it has to happen within the .then block as well.   

Now we are instantiating and doing our DOM manipulation used those instances and method all within the .thens.

Note that the previous dataset files are no longer being imported.  
All data is being pulled from the api but there are errors - see manual testing section below.

#### Where should the reviewer start?

The scripts file holds all the major changes.  You will see the rearrangement of the existing code to put it within the .then blocks. 

You will see that our user is being instantiated with a random number serving as the user.id rather than the index in the array. 

You will see that the `date` variable has been renamed `today` and is pulling the actual live date and reformatting it into a string that matches the dates in our dataset.  
All uses of the previous `date` variable have been replaced with our new `today` variable.

#### How should this be manually tested?

Pulling down and manually testing this PR's branch will be especially important because of the implementation of `fetch` and removal of the import tags for our old datasets. 

Error:
In the dev tools console, I am getting the console log that is set to our .catch for our hydration fetch - the last to be fetched of the 4.   'hydrationData error'

Today's hydration data (numOunces) is not displaying on the DOM.

However, the network tab shows 200 which indicates no error.  

If I keep refreshing, once in a while it will also console.log the error for activity data.  
Again. the network tab shows a status of 200 AND the activity data is still displaying on the DOM even when this error console.logs.  

I took screenshots but it will be best if you check it out yourself.  

#### Any background context you want to provide?
All noted under each section above.

#### What are the relevant tickets?

Issue #17 has been updated with more detail to address later as a low priority. It has a label of 'bug' since this logic could cause a bug with a different dataset but it is not a true bug at this time.

This PR Closes #45.

Issue #46 has been opened by this PR regarding the errors being console.log for hydration data despite the successful 200 status.

Questions:
- Does the knowledge base need an update?
There are some tricky parts regarding why we needed to place everything inside our .then blocks.  Please let me know if my explanations above are unclear.
